app/src/Service/FileUploader.php

```diff
@@ -20,9 +20,14 @@ public function __construct(private string $targetDirectory)
     public function upload(UploadedFile $file): string
     {
         $originalFilename = pathinfo($file->getClientOriginalName(), PATHINFO_FILENAME);
+        $extension = (($fileExtension = $file->guessExtension() )?
+            '.'.$fileExtension:
+            ""
+        );
+        $fullFilename = $originalFilename.$extension;
```

Comment:
Ça aurait mérité un commit dédié, voire on aurait pu faire une PR juste pour ça (sans être indispensable).

D’un point de vue lisibilité, même retour sur l’affectation de la variable `$fileExtension` + j’avais jamais vu de condition ternaire sur plusieurs lignes, j’avoue que ça m’a fait un nœud au cerveau 😅 

Je me pose la question de ce qu’on doit faire si `guessExtension()` retourne null : c’est probablement que le fichier n’est pas valide. Je pense qu’il est préférable de refuser un tel fichier.

Je propose un truc du genre :

```suggestion
        $fileExtension = $file->guessExtension();
        
        if (!$fileExtension) {
            return '';
        }

        $fullFilename = "{$originalFilename}.{$fileExtension}";
```

En vrai idéalement il faudrait lever une exception et la récupérer dans le controller pour afficher une erreur proprement à l’utilisateur. Mais là ça irait au-delà de cette PR, donc dans l’immédiat ça me suffirait.

Reply:
✅